cpp/src/arrow/compute/kernels/hash.h

```diff
@@ -38,6 +38,8 @@ class FunctionContext;
 /// values. Implementations should be thread-safe
 class ARROW_EXPORT HashKernel : public UnaryKernel {
  public:
+  // XXX why are those methods exposed?
+  virtual Status Reset() = 0;
```

Comment:
As I look at this again, it might be better to handle this through some other means than the HashKernel. But we also need to be able to maintain state in the kernels and invoke them on multiple array chunks, so that the results can share a common dictionary, etc.

Reply:
For CSV parsing we'll need the reconcile the types after the fact, as different dictionaries might be produced by different threads.